mtags/src/main/scala-3/scala/meta/internal/pc/PcDocumentHiglightProvider.scala

```diff
@@ -0,0 +1,213 @@
+              )
+            /**
+             * All select statements such as:
+             * val a = hello.<<b>>
+             */
+            case sel: Select if sought(sel.symbol) && !sel.span.isZeroExtent =>
+              highlights + new DocumentHighlight(
+                pos.withSpan(sel.nameSpan).toLSP,
+                DocumentHighlightKind.Read,
+              )
+            /* all definitions:
+             * def <<foo>> = ???
+             * class <<Foo>> = ???
+             * etc.
+             */
+            case df: NamedDefTree
+                if sought(
+                  df.symbol
+                ) && !df.span.isZeroExtent && !df.symbol.isSetter =>
+              highlights + new DocumentHighlight(
+                pos.withSpan(df.nameSpan).toLSP,
+                DocumentHighlightKind.Write,
+              )
+            /* Named parameters don't have symbol so we need to check the owner
+             * foo(<<name>> = "abc")
+             * User(<<name>> = "abc")
+             * etc.
+             */
+            case apply: Apply =>
+              val namedParam = apply.args.collectFirst {
+                case arg: NamedArg
+                    if sought.exists(sym =>
+                      sym.name == arg.name &&
+                        // foo(name = "123") for normal params
+                        (sym.owner == apply.symbol ||
+                          // Bar(name = "123") for case class, copy and apply methods
+                          apply.symbol.is(Flags.Synthetic) &&
+                          (sym.owner == apply.symbol.owner.companion || sym.owner == apply.symbol.owner))
+                    ) =>
+                  arg
+              }
+              namedParam match
+                case Some(arg) =>
+                  val realName = arg.name.stripModuleClassSuffix.lastPart
+                  highlights + new DocumentHighlight(
+                    pos
+                      .withSpan(
+                        arg.span.withEnd(arg.span.start + realName.length)
+                      )
+                      .toLSP,
+                    DocumentHighlightKind.Write,
+                  )
+
+                case None =>
+                  highlights
+            case mdf: MemberDef if mdf.mods.annotations.nonEmpty =>
+              val trees = collectTrees(mdf.mods.annotations)
+              val traverser =
+                new DeepFolder[Set[DocumentHighlight]](collectNames)
+              trees.foldLeft(highlights) { case (set, tree) =>
+                traverser(set, tree)
+              }
+            case o =>
+              highlights
+
+        val traverser = new DeepFolder[Set[DocumentHighlight]](collectNames)
+        val all = traverser(Set.empty[DocumentHighlight], unit.tpdTree)
+        all.toList.distinct
+      case None => Nil
+    end match
+  end higlights
+
+  // @note (tgodzik) Not sure currently how to get rid of the warning, but looks to correctly
+  @nowarn
```

Comment:
I was having some issue with getting things to work correctly with Positioned and untpd.Tree, I am not sure what the hierarchy is there, but this seems to work.

Anyone else had an issue like this?